bridge/client/app/project-board/project-board.component.ts

```diff
@@ -25,67 +25,79 @@ export class ProjectBoardComponent implements OnInit, OnDestroy {
 
   public error: string = null;
 
-  constructor(private _changeDetectorRef: ChangeDetectorRef, private router: Router, private route: ActivatedRoute, private dataService: DataService) { }
+  constructor(private router: Router, private route: ActivatedRoute, private dataService: DataService) { }
 
   ngOnInit() {
-    this.route.params
-      .pipe(takeUntil(this.unsubscribe$))
-      .subscribe(params => {
-        if (this.route.snapshot.url[0].path === 'trace') {
-          this.dataService.loadTracesByContext(params.shkeptncontext);
-          this.dataService.traces
-            .pipe(filter(traces => !!traces))
-            .pipe(takeUntil(this.unsubscribe$))
-            .subscribe((traces: Trace[]) => {
-              if(traces.length > 0) {
-                if(params.eventselector) {
-                  let trace = traces.find((t: Trace) => t.data.stage == params.eventselector && !!t.getProject() && !!t.getService());
-                  if (trace) {
-                    this.router.navigate(['/project', trace.getProject(),'sequence', trace.shkeptncontext, 'stage', trace.getStage()]);
-                  } else {
-                    trace = traces.reverse().find((t: Trace) => t.type == params.eventselector && !!t.getProject() && !!t.getService());
-                    if(trace) {
-                      this.router.navigate(['/project', trace.getProject(), 'sequence', trace.shkeptncontext, 'event', trace.id]);
-                    } else {
-                      this.error = "trace";
-                    }
-                  }
-                } else {
-                  let trace = traces.find((t: Trace) => !!t.getProject() && !!t.getService());
-                  this.router.navigate(['/project', trace.getProject(),'sequence', trace.shkeptncontext]);
-                }
-              } else {
-                this.error = "trace";
-              }
-            });
+    const projectName$ = this.route.params.pipe(
+      map(params => params.projectName),
+      filter(projectName => projectName)
+    );
+
+    this.project$ = projectName$.pipe(
+      switchMap(projectName => this.dataService.getProject(projectName))
+    );
```

Comment:
Same as the comment above, do we need the `const projectName$` if it is used only once or could we combine those lines in one `pipe()`?